src/main/java/com/nike/cerberus/operation/core/GenerateCertsOperation.java

```diff
@@ -0,0 +1,111 @@
+
+import static com.nike.cerberus.service.ConsoleService.DefaultAction.NO;
+
+/**
+ * Operation that uses the cert service to generate the certificates needed to enable https in a Cerberus Env.
+ */
+public class GenerateCertsOperation implements Operation<GenerateCertsCommand> {
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    private final CertificateService certService;
+    private final EnvironmentMetadata environmentMetadata;
+    private final ConsoleService consoleService;
+
+    @Inject
+    public GenerateCertsOperation(CertificateService certService,
+                                  EnvironmentMetadata environmentMetadata,
+                                  ConsoleService consoleService) {
+
+        this.certService = certService;
+        this.environmentMetadata = environmentMetadata;
+        this.consoleService = consoleService;
+    }
+
+    @SuppressFBWarnings(
+            value = "REC_CATCH_EXCEPTION",
+            justification = "I do not want to catch 1000 individual exceptions"
+    )
+    @Override
+    public void run(GenerateCertsCommand command) {
+        // The common name ex: demo.cerberus.io
+        String commonName = String.format("%s.%s", environmentMetadata.getName(), command.getBaseDomainName());
+        // The region specific subject alternate name. EX demo.us-west-2.ceberus.io
+        String regionSpecificSAN = String.format("%s.%s.%s",
+                environmentMetadata.getName(), environmentMetadata.getRegionName(), command.getBaseDomainName());
+
+        Set<String> subjectAlternativeNames = new HashSet<>();
+        subjectAlternativeNames.addAll(command.getSubjectAlternativeNames()); // add any additional SANs to the SAN set
+        subjectAlternativeNames.add(regionSpecificSAN); // add the default region specific san
+
+        // confirm with user
+        consoleService.askUserToProceed(String.format("Preparing to generate certs with Common Name: %s and Subject Alternative Names: %s",
+                        commonName, String.join(", ", subjectAlternativeNames)), NO);
+
+        // Enable the use of the hard coded lets encrypt cert if enabled
+        if (command.enableLetsEncryptCertfix()) {
+            log.warn("Setting acme4j.le.certfix system property to 'true', this only works if you use the special acme:// lets encrypt addr. See: https://shredzone.org/maven/acme4j/usage/session.html and https://shredzone.org/maven/acme4j/provider.html");
+            System.setProperty("acme4j.le.certfix", "true");
+        }
+
+        try {
+            // Use a temp dir or local dir if provided
+            File certDir = new File(command.getCertDir());
+
+            // check that we can write to the provided dir
+            FileUtils.forceMkdir(certDir);
+            if (! certDir.isDirectory() || ! certDir.canWrite()) {
+                throw new RuntimeException("Cert dir file is not a directory or is not writable, path: " + certDir.getAbsolutePath());
```

Comment:
"Certificate directory" or the param name

Reply:
- [ ] Done